src/pg_ext/big_decimal.cr

```diff
@@ -0,0 +1,23 @@
+require "big"
+
+module PG
+  struct Numeric
+    # Returns a BigDecimal representation of the numeric. This retains all precision.
+    def to_big_d
+      return BigDecimal.new("0") if nan? || ndigits == 0
+
+      # Since BigDecimal allows initialaztion from String, why should one reinvent the wheel?
+      BigDecimal.new(to_s)
```

Comment:
Thanks for the note here, I don't think the comment should remain in the codebase though, this method could become:
```crystal
def to_big_d
  return BigDecimal.new("0") if nan? || ndigits == 0
  BigDecimal.new(to_s)
end
```

Does anyone know the overhead of calling `to_s` here? Is there another more efficient overload for `BigDecimal.new` ?

Reply:
At least I didn't find a better way to initialize BigDecimal, without potentially loosing precision by using `BigDecimal.new(to_f)`